File: 1957-delete-characters-to-make-fancy-string/1957-delete-characters-to-make-fancy-string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        """
        Modifies the input string `s` to a "fancy" string where no character appears more than twice in a row.

        Parameters:
        s (str): The input string containing lowercase English letters.

        Returns:
        str: A fancy string with no character appearing more than twice consecutively.
        """
        result = ""  # Initialize the result string to build the fancy string
        count = 0  # Counter to track consecutive occurrences of the current character

        # Iterate over each character in the input string
        for ch in s:
            if not result or ch != result[-1]:  # Start counting for new characters
                count = 1
            else:
                count += 1  # Increment counter for consecutive characters
            # Add character only if it appears less than three times consecutively
            if count < 3:
                result += ch

        return result

File: Difficulty/A. Easy/1957-delete-characters-to-make-fancy-string/1957-delete-characters-to-make-fancy-string.c
#include <stdlib.h>
#include <string.h>

/**
 * @brief Creates a fancy string from the input by ensuring no character appears more than twice consecutively.
 *
 * @param s (char*): The input string consisting of lowercase English letters.
 *
 * @return (char*): A dynamically allocated fancy string with no character appearing more than twice consecutively.
 */
char* makeFancyString(char* s) {
    int n = strlen(s);
    char* result = (char*)malloc((n + 1) * sizeof(char)); // Allocate space for the resulting string
    int count = 1; // To count consecutive identical characters
    int j = 0;     // Index for filling the result string

    // Ensure input is valid
    if (s == NULL) return NULL;

    // Add the first character to result
    result[j++] = s[0];

    // Process each character starting from the second character
    for (int i = 1; i < n; i++) {
        if (s[i] == s[i - 1]) {
            count++; // Increment count for consecutive identical characters
        } else {
            count = 1; // Reset count if character is different
        }

        // Append character if it appears less than 3 times consecutively
        if (count < 3) {
            result[j++] = s[i];
        }
    }

    result[j] = '\0'; // Null-terminate the result string
    return result;
}
